pkg/operations/armhelpers/azureclients.go

```diff
@@ -0,0 +1,32 @@
+package armhelpers
+
+import (
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// AzureClients contains Azure clients
+type AzureClients struct {
+	SubscriptionID   string
+	AzureEnvironment azure.Environment
+	TenantID         string
+
+	GroupsClient *resources.GroupsClient
+	VMClient     *compute.VirtualMachinesClient
+}
+
+// Create method creates various Azure clients
+func (ac *AzureClients) Create(token *adal.ServicePrincipalToken) (*AzureClients, error) {
```

Comment:
Generally, the ctor style in Golang would be to make this:

`func NewAzureClient() (*AzureClients, error) {` or maybe, but less likely:

`func (ac *AzureClients) Initiailize(token) error {`

as is, this is fairly un-idiotmatic since it returns itself and is a pointer receiver function at the same time.

I was expecting to see the `Create` called "higher up" and then attached to the upgradeCluster operation before the upgradeCluster operation is kicked off.